2004-10-18  Maciej W. Rozycki  <macro@mips.com>

	* sysdeps/mips/bits/dlfcn.h (RTLD_DEEPBIND): Define.

 Please apply.

  Maciej

glibc-2.3.3-20041018-mips-rtld_deepbind-2.patch
diff -up --recursive --new-file glibc-2.3.3-20041018.macro/sysdeps/mips/bits/dlfcn.h glibc-2.3.3-20041018/sysdeps/mips/bits/dlfcn.h
--- glibc-2.3.3-20041018.macro/sysdeps/mips/bits/dlfcn.h	Sat Oct 27 02:25:51 2001
+++ glibc-2.3.3-20041018/sysdeps/mips/bits/dlfcn.h	Mon Oct 18 16:21:44 2004
@@ -1,5 +1,6 @@
 /* System dependent definitions for run-time dynamic loading.
-   Copyright (C) 1996, 1997, 1999, 2000, 2001 Free Software Foundation, Inc.
+   Copyright (C) 1996, 1997, 1999, 2000, 2001, 2004
+	Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -24,8 +25,9 @@
 /* The MODE argument to `dlopen' contains one of the following: */
 #define RTLD_LAZY	0x0001	/* Lazy function call binding.  */
 #define RTLD_NOW	0x0002	/* Immediate function call binding.  */
-#define	RTLD_BINDING_MASK  0x3	/* Mask of binding time value.  */
+#define RTLD_BINDING_MASK  0x3	/* Mask of binding time value.  */
 #define RTLD_NOLOAD	0x00008	/* Do not load the object.  */
+#define RTLD_DEEPBIND	0x00010	/* Use deep binding.  */
 
 /* If the following bit is set in the MODE argument to `dlopen',
    the symbols of the loaded object and its dependencies are made
